presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analyzer.java

```diff
@@ -88,7 +90,7 @@ public Analysis analyze(Statement statement, boolean isDescribe)
                                 accessControlInfo.getIdentity(),
                                 session.getAccessControlContext(),
                                 tableName,
-                                columns)));
+                                isCheckAccessControlAfterColumnPruning(session) ? analysis.getTableColumnReferencesAfterPruning().get(tableName) : columns)));
```

Comment:
Add a helper function in Analysis that takes a table and returns columns so all the access checking logic is in one place.

Reply:
Done